src/Symfony/Component/Console/Application.php

```diff
@@ -121,18 +122,23 @@ public function run(InputInterface $input = null, OutputInterface $output = null
 
         try {
             $exitCode = $this->doRun($input, $output);
-        } catch (\Exception $e) {
+        } catch (\Exception $runException) {
```

Comment:
there *may* be something to enhance on the console component when dealing with \Error exception, but I'd suggest doing it in a dedicated PR on master